plugins/CircuitExplorer/api/neuron/processors/SectionSmoother.cpp

```diff
@@ -0,0 +1,133 @@
+    {
+        auto next = _smoothQueue.front();
+        _smoothQueue.pop_front();
+        return next;
+    }
+
+    bool empty() const noexcept
+    {
+        return _smoothQueue.empty();
+    }
+
+private:
+    std::deque<size_t> _smoothQueue;
+};
+
+class ParentChildrenSmoother
+{
+public:
+    ParentChildrenSmoother(float somaRadiusMultiplier, float sectionRadiusChange)
+        : _somaRadiusMultiplier(somaRadiusMultiplier)
+        , _sectionRadiusChange(sectionRadiusChange)
+    {
+        assert(somaRadiusMultiplier > 0.f);
+        assert(_sectionRadiusChange >= 0.f && _sectionRadiusChange <= 1.f);
+    }
+
+    void apply(NeuronMorphology &morphology)
+    {
+        SmoothQueue queue;
+        _initializeRootSections(morphology, queue);
+
+        while (!queue.empty())
+        {
+            auto parentIndex = queue.next();
+            auto parentSection = _getSection(morphology, parentIndex);
+            auto parentRadius = parentSection.samples.front().radius;
+
+            auto childSections = morphology.sectionChildrenIndices(parentSection.id);
+            auto childrenRadius = parentRadius * (1.f - _sectionRadiusChange);
+            _setSectionRadius(morphology, childSections, childrenRadius);
+
+            queue.append(childSections);
+        }
+    }
+
+private:
+    void _initializeRootSections(NeuronMorphology &morphology, SmoothQueue &queue)
+    {
+        auto somaChildren = morphology.sectionChildrenIndices(-1);
+        queue.append(somaChildren);
+        if (!morphology.hasSoma())
+        {
+            return;
+        }
+
+        auto startingRadius = morphology.soma().radius * _somaRadiusMultiplier;
+        _setSectionRadius(morphology, somaChildren, startingRadius);
+    }
+
+    void _setSectionRadius(NeuronMorphology &morphology, const std::vector<size_t> &sectionIndices, float radius)
+    {
+        auto &sections = morphology.sections();
+        for (auto sectionIndex : sectionIndices)
+        {
+            auto &section = sections[sectionIndex];
+            for (auto &sample : section.samples)
+            {
+                sample.radius = radius;
+            }
+        }
+    }
+
+    const NeuronMorphology::Section &_getSection(const NeuronMorphology &morphology, size_t sectionIndex)
+    {
+        auto &sections = morphology.sections();
+        auto &section = sections[sectionIndex];
+        return section;
+    }
+
+    float _getSectionRadius(const NeuronMorphology &morphology, size_t sectionIndex)
+    {
+        auto &sections = morphology.sections();
+        auto &section = sections[sectionIndex];
+        auto &samples = section.samples;
+        return samples.front().radius;
+    }
+
+private:
+    float _somaRadiusMultiplier;
+    float _sectionRadiusChange;
+};
+} // namespace
+
+void SectionSmoother::process(NeuronMorphology &morphology) const
+{
+    constexpr float somaRadiusMultiplier = 0.25f;
+    constexpr float sectionRadiusChange = 0.11f;
```

Comment:
Actually I just found on YouTube that it is better to use static constexpr for constexpr locals so you don't push it on the stack on each function call.

Reply:
Good point, however these parameters will be soon made availble through the morphology loader, so Ill avoid the recommit.